compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -126,6 +131,7 @@ impl Context {
         instruction_id: InstructionId,
         dfg: &DataFlowGraph,
         constrained_values: &mut HashMap<ValueId, ValueId>,
+        side_effects_enabled: &bool,
```

Comment:
```suggestion
        side_effects_enabled: bool,
```

Let's dereference this boolean earlier